lib/src/rules/invariant_booleans.dart

```diff
@@ -224,6 +227,16 @@ AstNodePredicate _noFurtherAssignmentInvalidatingCondition(
       0;
 }
 
+Iterable<Expression> _splitConjunctions(Expression expression) {
+  if (expression is BinaryExpression &&
+      expression.operator.type == TokenType.AMPERSAND_AMPERSAND) {
+    return _splitConjunctions(expression.leftOperand).toList()
```

Comment:
If you return `List<Expression>`, then you won't need the `toList` here and can remove some unnecessary list copying operations.